pkg/asset/installconfig/platformprovisioncheck.go

```diff
@@ -83,6 +84,19 @@ func (a *PlatformProvisionCheck) Generate(ctx context.Context, dependencies asse
 		if err != nil {
 			return err
 		}
+		isAro := false
+		if ic.Config.Azure.ResourceGroupName != "" {
+			isAro, err = client.CheckIfARO(ctx, ic.Config.Azure.ResourceGroupName)
+			if err != nil {
+				return err
+			}
+		}
+
+		if !isAro && ic.Config.Publish != types.InternalPublishingStrategy {
+			if ic.Config.Azure.BaseDomainResourceGroupName == "" {
```

Comment:
It's not clear to me why ARO wants to be able to create an non-private cluster without a BaseDomainResourceGroupName. I think this will for sure cause problems in phase 3, so we might as well look into it now.